src/main/java/com/dataguise/saas/service/TaskBuilderService.java

```diff
@@ -0,0 +1,148 @@
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    maskerTaskBuilder.addPolicy(8);
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) maskerTaskBuilder;
+        } else if( taskBuilder instanceof LFADetectionTask.LFADetectionTaskBuilder) {
+            LFADetectionTask.LFADetectionTaskBuilder LFADetectionTaskBuilder = ((LFADetectionTask.LFADetectionTaskBuilder) taskBuilder);
+            for(String policyType : policyList) {
+                if(policyType.equalsIgnoreCase(PolicyType.HIPAA.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("HIPAA_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.PCI.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("PCI_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.PII.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("PII_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("GDPR_Hadoop");
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) LFADetectionTaskBuilder;
+        }
+        return builder;
+    }
+    public String getTimeStampForDgSecure() {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+        Date date = new Date();
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        sdf.setTimeZone(tz);
+        String timeStamp = sdf.format(date);
+
+        logger.info("TIME STAMP IS: " + timeStamp);
+        return timeStamp;
+    }
+
+
+    public List<SensitiveExpressionDTO> getSensitiveExpressionDetails(Optional<Integer> regexId) throws BadGatewayException, IOException {
+        List<SensitiveExpressionDTO> sensitiveExpressionList = null;
+        try {
+            String sensitiveExpressionDetails =  restDriver.getSensitiveExpressionDetails();
+            ObjectMapper mapper = new ObjectMapper();
+
+            TypeReference<List<SensitiveExpressionDTO>> mapType = new TypeReference<List<SensitiveExpressionDTO>>(){};
+            sensitiveExpressionList = mapper.readValue(sensitiveExpressionDetails, mapType);
+
+            if (regexId.isPresent()) {
+                sensitiveExpressionList = sensitiveExpressionList.stream().
+                        filter(sensitiveExpressionDTO -> sensitiveExpressionDTO.getRegexId().equals(regexId.get()))
+                        .collect(Collectors.toList());
+            }
+        } catch (DgDrvException e) {
+            logger.error("Error in getting sensitive details: "+e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/TaskBuilderService.java L143
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1571044579>*